rero_ils/modules/items/api/api.py

```diff
@@ -157,25 +160,41 @@ def get_item_record_for_ui(cls, **kwargs):
         return item
 
     @classmethod
-    def get_items_with_obsolete_temporary_item_type(cls, end_date=None):
-        """Get all items with an obsolete temporary item_type.
+    def format_end_date(cls, end_date):
+        """Return a formatted end date."""
+        # (`datetime.now(timezone.utc)` by default)
+        if end_date is None:
+            end_date = datetime.now(timezone.utc)
+        end_date = end_date.strftime('%Y-%m-%d')
+        return end_date
+
+    @classmethod
+    def get_items_with_obsolete_temporary_item_type_or_location(
+            cls, end_date=None):
+        """Get all items with an obsolete temporary item_type or location.
 
-        An end_date could be attached to the item temporary item_type. If this
-        date is less or equal to sysdate, then the temporary_item_type must be
-        considered as obsolete.
+        An end_date could be attached to the item temporary item_type or
+        temporary location. If this date is less or equal to sysdate, then the
+        temporary_item_type or temorary_location must be considered as obsolete
+        and the field must be removed.
 
-        :param end_date: the end_date to check (`datetime.now()` by default)
+        :param end_date: the end_date to check.
         :return A generator of `ItemRecord` object.
         """
-        if end_date is None:
-            end_date = datetime.utcnow()
-        end_date = end_date.strftime('%Y-%m-%d')
-        results = ItemsSearch() \
-            .filter('range', temporary_item_type__end_date={'lte': end_date}) \
-            .source('pid') \
-            .scan()
-        for result in results:
-            yield Item.get_record_by_pid(result.pid)
+        end_date = cls.format_end_date(end_date)
+        items_query = ItemsSearch()
+        loc_es_quey = items_query.filter(
+                    'range', temporary_location__end_date={'lte': end_date})
+        locs = [
+            (hit.meta.id, 'loc') for hit in loc_es_quey.source('pid').scan()]
+
+        itty_es_query = items_query.filter(
+                'range', temporary_item_type__end_date={'lte': end_date})
+        itty = [(hit.meta.id, 'itty') for hit in itty_es_query.source(
+            'pid').scan()]
+        hits = itty + locs
+        for id, field_type in hits:
```

Comment:
`field_type` is not used. We can use `_` or we we not saving the field type during creation.